README.md

```diff
@@ -11,138 +36,172 @@ Our smart contracts act as a **bridge** between various blockchains (like Linea,
 ✅ **Filecoin (Storage Destination)**
 - **`DealClientAxl`** – Receives data & initiates storage deals
 
----
-
-## **📦 Installation & Setup**
-
-### **1️⃣ Clone & Install Dependencies**
-```sh
+## Architecture
+
+The system works through three main components deployed across chains:
+
+1. **Source Chain Contracts**: 
+   - OnRampContract from `Onramp.sol`: Manages data submissions and payment
+   - AxelarBridge from `Oracle.sol`: Handles cross-chain communication
+2. **Destination Chain Contract (Filecoin)**:
+   - DealClientAxl from `Prover-Axelar.sol`: Processes storage deals and manages data verification
+3. **xChain Client**: Coordinates the data transfer and deal-making process
+
+## Project Structure
```

Comment:
Let's remove this section. Structure is going to be changed over time and it is not very helpful at this stage.

Reply:
Alright